mobile/test/cc/engine_test.cc

```diff
@@ -51,32 +52,38 @@ TEST(EngineTest, SetLogger) {
                     })
                     .start();
 
-  auto request_headers =
-      Platform::RequestHeadersBuilder(Platform::RequestMethod::GET, "https", "example.com", "/")
-          .build();
+  auto request_headers = Platform::RequestHeadersBuilder(
+                             Platform::RequestMethod::GET, "https",
+                             absl::StrFormat("localhost:%d", test_server.getServerPort()), "/")
+                             .build();
   stream->sendHeaders(std::make_shared<Platform::RequestHeaders>(request_headers), true);
   stream_complete.WaitForNotification();
 
+  // It is important that the we shutdown the TestServer first before terminating the Engine. This
+  // is because when the Engine is terminated, the Logger::Context will be destroyed and
+  // Logger::Context is a global variable that is used by both Engine and TestServer. By shutting
+  // down the TestServer first, the TestServer will no longer access a Logger::Context that has been
+  // destroyed.
```

Comment:
This is a bit brittle (will require everyone to see this comment and remember to do this).  Is it possible to wrap this into a test class, e.g. `EngineWithServer` which takes an EngineBuilder in its constructor, starts up the test server and Engine, has an accessor method to allow the caller to access the Engine, and then on destruction, shuts down the TestServer and Engine in the correct order.

Reply:
That's a great suggestion. Done.